src/content/ui/adaptive-responsive/layout.md

```diff
@@ -0,0 +1,494 @@
+---
+title: Layout for adaptive apps
+description: >
+  Learn how to create a responsive app >
+  that responds to changes in the screen size.
+short-title: Layout
+---
+
+<?code-excerpt path-base="ui/adaptive_app_demos"?>
+
+<b>TODO(sfshaza): I'm coalescing this page from two previous pages. I'm not done with the text. It likely also needs code or other updates.</b>
+
+Flutter allows you to create apps that self-adapt
+to the device's screen size and orientation.
+
+## General advice
+
+When creating a layout that automatically adapts
+to the size and shape of the user's device,
+there are several things you might keep in mind.
+This page covers how to take an app build for a
+phone and make it beautiful on a wide range of devices.
+
+## Two approaches
```

Comment:
I think this page should follow the flow outlined here https://docs.google.com/presentation/d/1DKsQtmEUo1HaVyHsqYy1fBj8NwdQyxS4PtsiRnlKZyk/edit#slide=id.g2c2f93baf95_3_626

I dont think MediaQuery.of and LayoutBuilder are approaches. They are more like tools or maybe better put they are tools you use once you have decided if you want the space to change based on the size of the window or based on the size of the area a widget can use.

Reply:
Yeah I think this section could instead be called "Approach" or something like that, and then outline that 3 step "Abstract, Measure, Branch" methodology.

Having a code example underneath from something like our dialog example in the slides would be nice and concise as well.

Reply:
I really like this idea! having user decide what it is they are intending to do and then giving them the right tools/approach to do it